rasa/core/actions/forms.py

```diff
@@ -348,6 +360,10 @@ def _add_dynamic_slots_requested_by_dynamic_forms(
     def _get_slot_extractions(
         self, tracker: "DialogueStateTracker", domain: Domain,
     ) -> Dict[Text, Any]:
+        if not tracker.active_loop:
+            # pre-filled slots were already validated at form activation
+            return {}
+
```

Comment:
Hm, `_get_slot_extractions` is only called by `validate` which is invoked by `_validate_if_required` if there's no active loop set (i.e. we're in that activation case) or we're in a loop -- If I interpret this correctly, then we might want to pull this check into `_validate_if_required` and explicitly *not* call any validate there in that second case?

Reply:
Not sure I understood this part, do you mean breaking [`needs_validation`](https://github.com/RasaHQ/rasa/blob/2c17cd24003e2cc9e23276e02b32757445867774/rasa/core/actions/forms.py#L527) into 2 checks, one for `not tracker.active_loop` in which case we return `[ ]` and the other half of the `or` check which can be used in the `if/else` [here](https://github.com/RasaHQ/rasa/blob/2c17cd24003e2cc9e23276e02b32757445867774/rasa/core/actions/forms.py#L532)?

Reply:
yep